xmppserver/src/main/java/org/jivesoftware/openfire/net/RespondingServerStanzaHandler.java

```diff
@@ -92,12 +104,23 @@ protected boolean isStartOfStream(String xml) {
             // Pull namespaces off of the stream:stream stanza and add them to the additional
             List<Namespace> receivedNamespaces = null;
             try {
-                receivedNamespaces = DocumentHelper.parseText(xml + "</stream:stream>").getRootElement().declaredNamespaces();
+                Element rootElement = DocumentHelper.parseText(xml + "</stream:stream>").getRootElement();
+                receivedNamespaces = rootElement.declaredNamespaces();
                 Set<Namespace> additionalNamespaces = receivedNamespaces
                     .stream()
                     .filter(RespondingServerStanzaHandler::isRelevantNamespace)
                     .collect(Collectors.toSet());
                 connection.setAdditionalNamespaces(additionalNamespaces);
+
+                // Create a new session with a new ID if a new stream has started on an existing connection
```

Comment:
Maybe add a comment to explain why the new session is created. Future people maintaining this code will need to understand that.

Reply:
Added